go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,385 @@
+	}
+
+	return event
+}
+
+type Ampli struct {
+	Disabled bool
+	Client   amplitude.Client
+	mutex    sync.Mutex
+}
+
+// Load initializes the Ampli wrapper.
+// Call once when your application starts.
+func (a *Ampli) Load(options LoadOptions) {
+	a.mutex.Lock()
+	a.Disabled = options.Disabled
+	a.mutex.Unlock()
+
+	if a.Client != nil {
+		log.Default().Printf("Warn: Ampli is already initialized. Ampli.load() should be called once at application start up.")
+
+		return
+	}
+
+	if options.Client.isEmpty() {
+		options.Client = LoadClientOptions{
+			Configuration: DefaultConfiguration(),
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Instance.APIKey or APIKey[options.Environment], " +
+			"or options.Instance.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration()
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration().Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+	return !a.Disabled && a.Client != nil
+}
+
+func (a *Ampli) setUserID(userID string, eventOptions *amplitude.EventOptions) {
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+}
+
+// Track tracks an StronglyTypedEvent.
+func (a *Ampli) Track(userID string, event StronglyTypedEvent, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	a.setUserID(userID, &eventOptions)
+
+	baseEvent := event.ToEvent()
+	baseEvent.EventOptions = eventOptions
+
+	a.Client.Track(baseEvent)
+}
+
+// Identify identifies a user and set user properties.
+func (a *Ampli) Identify(userID string, identify *Identify, eventOptions amplitude.EventOptions) {
+	a.Track(userID, identify, eventOptions)
+}
+
+// GroupIdentify identifies a group and set group properties.
+func (a *Ampli) GroupIdentify(groupType string, groupName string, group *Group, eventOptions amplitude.EventOptions) {
+	group.groups = map[string][]string{groupType: {groupName}}
+
+	a.Track("", group, eventOptions)
+}
+
+// SetGroup sets group for the current user.
+func (a *Ampli) SetGroup(userID string, groupType string, groupName []string, eventOptions amplitude.EventOptions) {
```

Comment:
Do we need userID for the setGroup?

Reply:
I think for server side SDKs need to accept userId since different events may belong to different users. But for client side SDKs, it is not necessary as all events are from the same user.

More discussion: https://amplitude.slack.com/archives/C02E1H87E2C/p1660948525608159?thread_ts=1660947850.763319&cid=C02E1H87E2C